server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/CacheContainerConfigurationBuilder.java

```diff
@@ -290,15 +290,16 @@ private ClassLoader makeGlobalClassLoader(ModuleLoader moduleLoader, ModuleIdent
                 classLoaders.add(moduleLoader.loadModule(additionalModule).getClassLoader());
             }
         }
-        switch (classLoaders.size()) {
-            case 0:
-                // default CL
-                return CacheContainerConfiguration.class.getClassLoader();
-            case 1:
-                return classLoaders.iterator().next();
-            default:
-                return new AggregatedClassLoader(classLoaders);
+        ClassLoader infinispanSubsystemClassloader = CacheContainerConfiguration.class.getClassLoader();
+        if (classLoaders.isEmpty()) {
+            // use our default CL if nothing was specified by user
+            return infinispanSubsystemClassloader;
         }
+        if (cacheContainerModule == null) {
+            // do not use the infinispan subsystem classloader if the user specifically requested a different one
+            classLoaders.add(infinispanSubsystemClassloader);
+        }
+        return new AggregatedClassLoader(classLoaders);
```

Comment:
When running the server, ```AggregatedClassLoader``` will always be created, since ```cacheContainerModule``` will always be ```org.infinispan.extension:ispn-9.4```. Should we return instead ```moduleLoader.loadModule(cacheContainerModule).getClassLoader()``` when there is no ```additionalModules```?

Reply:
if ```classLoaders``` has just one element we could return that instead of making an AggregatedClassLoader

Reply:
done